SU2_CFD/src/numerics/flow/flow_sources.cpp

```diff
@@ -729,19 +731,41 @@ CNumerics::ResidualType<> CSourceIncStreamwise_Periodic::ComputeResidual(const C
   /*--- Compute the periodic temperature contribution to the energy equation, if energy equation is considered ---*/
   if (energy && streamwisePeriodic_temperature) {
 
-    scalar_factor = SPvals.Streamwise_Periodic_IntegratedHeatFlow * DensityInc_i / (SPvals.Streamwise_Periodic_MassFlow * norm2_translation);
+    /*--- Compute scalar factors, if there is a HEAT FLUX BC ---*/
+    if (bool_heat_flux_bc) {
+      scalar_factor = SPvals.Streamwise_Periodic_IntegratedHeatFlow * DensityInc_i / (SPvals.Streamwise_Periodic_MassFlow * norm2_translation);
 
-    /*--- Compute scalar-product dot_prod(v*t) ---*/
-    dot_product = GeometryToolbox::DotProduct(nDim, Streamwise_Coord_Vector, &V_i[1]);
+      /*--- Compute scalar-product dot_prod(v*t) ---*/
+      dot_product = GeometryToolbox::DotProduct(nDim, Streamwise_Coord_Vector, &V_i[1]);
+    }
+
+    /*--- Compute scalar factors, if there is an ISOTHERMAL BC ---*/
+    if (bool_isotherml_bc) {
+
+      /*--- Compute three terms associated with the source terms for iso-thermal BCs ---*/
+      // Reference Eq(20) Stalio et. al, doi:10.1115/1.2717235
+
+      dot_product = GeometryToolbox::DotProduct(nDim, Streamwise_Coord_Vector, PrimVar_Grad_i[3]);
+      su2double term1 = 2 * Thermal_Conductivity_i * dot_product;
+      su2double term2 = - SPvals.Streamwise_Periodic_LambdaL * Thermal_Conductivity_i * V_i[3];
+      su2double term3 = - V_i[3] * V_i[1] * DensityInc_i *  config->GetSpecific_Heat_Cp();
+      scalar_factor = SPvals.Streamwise_Periodic_LambdaL * (term1 + term2 + term3);
+      dot_product = 1.0;
+    }
 
     residual[nDim+1] = Volume * scalar_factor * dot_product;
 
     /*--- If a RANS turbulence model is used, an additional source term, based on the eddy viscosity gradient is added. ---*/
     if(turbulent) {
-
-      /*--- Compute a scalar factor ---*/
-      scalar_factor = SPvals.Streamwise_Periodic_IntegratedHeatFlow / (SPvals.Streamwise_Periodic_MassFlow * sqrt(norm2_translation) * Prandtl_Turb);
-
+      if (bool_heat_flux_bc) {
+        /*--- Compute a scalar factor ---*/
+        scalar_factor = SPvals.Streamwise_Periodic_IntegratedHeatFlow / (SPvals.Streamwise_Periodic_MassFlow * sqrt(norm2_translation) * Prandtl_Turb);
+      }
+      
+      if (bool_isotherml_bc) {
```

Comment:
What do you get if you have both?

Reply:
Good observation. Since the assumptions are completely different for the two BCs, they are not valid in the same fluid domain. An option could be to have a flag in option config to check that there is only type of BCs mentioned when SWP solver is active. Please let me know if you have other suggestions.

Reply:
Throwing an error in CConfig should be ok